hooks/gen_docs.py

```diff
@@ -1,10 +1,263 @@
 """Generates the whole 'generatable' KPOps documentation"""
+import os
 import subprocess
+from pathlib import Path
+
+import yaml
 
 from hooks import PATH_ROOT
+from kpops.cli.registry import _find_classes
+from kpops.components.base_components.pipeline_component import PipelineComponent
+from kpops.utils.yaml_loading import load_yaml_file
 
 PATH_KPOPS_MAIN = PATH_ROOT / "kpops/cli/main.py"
 PATH_CLI_COMMANDS_DOC = PATH_ROOT / "docs/docs/user/references/cli-commands.md"
+PATH_DOCS_RESOURCES = PATH_ROOT / "docs/docs/resources"
+PATH_DOCS_COMPONENTS = PATH_DOCS_RESOURCES / "pipeline-components"
+PATH_DOCS_COMPONENTS_DEPENDENCIES = (
+    PATH_DOCS_COMPONENTS / "dependencies/pipeline_component_dependencies.yaml"
+)
+PATH_DOCS_COMPONENTS_DEPENDENCIES_DEFAULTS = (
+    PATH_DOCS_COMPONENTS / "dependencies/defaults_pipeline_component_dependencies.yaml"
+)
+PATH_DOCS_KPOPS_STRUCTURE = PATH_DOCS_COMPONENTS / "dependencies/kpops_structure.yaml"
+KPOPS_COMPONENTS = tuple(_find_classes("kpops.components", PipelineComponent))
+KPOPS_COMPONENTS_INHERITANCE_REF: dict = {
+    component.get_component_type(): component.__base__.get_component_type()  # type: ignore[reportGeneralTypeIssues]
+    for component in KPOPS_COMPONENTS
+}
+KPOPS_COMPONENTS_FIELDS = {
+    # `set` to make it pickleable
+    component.get_component_type(): set(component.__fields__.keys())
+    for component in KPOPS_COMPONENTS
+}
+SECTIONS_ORDER = [
```

Comment:
👍🏻 for fixed ordering

Reply:
The only drawback is we would have to maintain it manually. I think the dev docs are becoming more and more important.

Reply:
how about getting the fields in the order they are defined?

```python
KPOPS_COMPONENTS_FIELDS = {
    component.get_component_type(): [
        field_name
        for field_name, model in component.__fields__.items()
        if not model.field_info.exclude
    ]
    for component in KPOPS_COMPONENTS
}
SECTION_ORDER = deduplicate(
    [field for fields in KPOPS_COMPONENTS_FIELDS.values() for field in fields]
)
```

I am currently tweaking the order in #290

Reply:
Would be a great improvement :)